sdk/spring/azure-spring-boot-starter-keyvault-certificates/src/main/java/com/azure/spring/security/keyvault/certificates/starter/KeyVaultCertificatesEnvironmentPostProcessor.java

```diff
@@ -24,9 +27,12 @@
 @Order
 public class KeyVaultCertificatesEnvironmentPostProcessor implements EnvironmentPostProcessor {
 
+    private static final DeferredLog LOGGER = new DeferredLog();
```

Comment:
the class should have a constructor to take DefferedLog as argument, should not create a new instance of deferred log, see `org.springframework.boot.env.EnvironmentPostProcessor ` javadoc